### PR TITLE
Rebranding of CBL-Mariner into Azure Linux in tools and configs

### PR DESCRIPTION
**Pull request by mbykhovtsev-ms**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

###### Change Log  

- Replaced hostname of produced images from `cbl-mariner` to `azurelinux`
- Replaced references to the github repo links from old to new
- Replaced go module from `CBL-Mariner` to `azurelinux` and references to it in imports of go tools
- Replaced references to `CBL-Mariner` in comments to `Azure Linux`

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: toolkit/scripts/containerized-build/resources/welcome.txt [hunk 1]
```diff
@@ -21,10 +21,10 @@
                                                         create new folder under /usr/src/mariner/SPECS
                                                         with .spec (and other files)
 
-*     Changes to /usr/src/mariner/SPECS will be available on host machine at <CBL-Mariner>/SPECS
+*     Changes to /usr/src/mariner/SPECS will be available on host machine at <Azure Linux>/SPECS
```
Comment (jslobodzian) [inline toolkit/scripts/containerized-build/resources/welcome.txt:24]: I believe this should read "azurelinux"/SPECS.  Not "Azure Linux"/SPECS

Same with the subsequent change

**Review by jslobodzian — APPROVED** [review]

File: toolkit/scripts/containerized-build/resources/welcome.txt (hunk 1 above)
Comment (mbykhovtsev-ms) [reply toolkit/scripts/containerized-build/resources/welcome.txt]: Fixed it

File: toolkit/scripts/tools.mk [hunk 2]
```diff
@@ -91,10 +91,10 @@ clean-go-tools:
 	rm -rf $(TOOL_BINS_DIR)
 	rm -rf $(BUILD_DIR)/tools
 
-go_ldflags := 	-X github.com/microsoft/CBL-Mariner/toolkit/tools/internal/exe.ToolkitVersion=$(RELEASE_VERSION) \
-				-X github.com/microsoft/CBL-Mariner/toolkit/tools/pkg/imagecustomizerlib.ToolVersion=$(IMAGE_CUSTOMIZER_FULL_VERSION) \
-				-X github.com/microsoft/CBL-Mariner/toolkit/tools/internal/exe.DistroNameAbbreviation=$(DIST_NAME_ABRV) \
-				-X github.com/microsoft/CBL-Mariner/toolkit/tools/internal/exe.DistroMajorVersion=$(dist_major_version_number)
+go_ldflags := 	-X github.com/microsoft/azure/toolkit/tools/internal/exe.ToolkitVersion=$(RELEASE_VERSION) \
```
Comment (neha170) [inline toolkit/scripts/tools.mk:94]: should this be azurelinux

Comment (neha170) [reply toolkit/scripts/tools.mk:94]: fixed in https://github.com/microsoft/azurelinux/pull/8195

**Pull request merged**